chunky/src/java/se/llbit/chunky/model/CauldronModel.java

```diff
@@ -412,6 +412,7 @@ public static boolean intersectWithWater(Ray ray, boolean stillWater, int level)
     Quad water = waterLevels[level];
     if (water != null && water.intersect(ray)) {
       if (!stillWater) {
+        // TODO Hook into the more generic water shading?
```

Comment:
Not sure if overkill, but you _could_ pass the `scene` down into this method to get access to `scene.getWaterShading()`.

Reply:
Forgot that intersect already has the scene, thought this would mean changing the signature of the intersect function of Block (and every derived class) but it only requires minimal change so I'll do it